sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,221 @@
+            else:
+                method = 'predict'
+        else:
+            method = self.method
+
+        return method
+
+    def transform(self, *args, **kwargs):
+        """Transform dataset.
+
+        Note that, unlike ``fit_transform()``, this won't return the cross
+        validation predictions. See the :ref:`User Guide <ensemble>` for more
+        info.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Input data to be transformed. Use ``dtype=np.float32`` for maximum
+            efficiency. Sparse matrices are also supported, use sparse
+            ``csr_matrix`` for maximum efficiency.
+
+        Returns
+        -------
+        X_transformed : sparse matrix, shape=(n_samples, n_out)
+            Transformed dataset.
+
+        """
+        t = getattr(self.base_estimator, self._method_name())
+        preds = t(*args, **kwargs)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+    def fit_transform(self, X, y, **fit_params):
+        """Fit estimator and transform dataset.
+
+        Note that this behavior is different from ``fit().transform()`` as it
+        will return the cross validation predictions instead.. See the
+        :ref:`User Guide <ensemble>` for more info.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Input data used to build forests. Use ``dtype=np.float32`` for
+            maximum efficiency.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        X_transformed : sparse matrix, shape=(n_samples, n_out)
+            Transformed dataset.
+        """
+        preds = cross_val_predict(self.base_estimator, X, y, cv=self.cv,
+                                  method=self._method_name(),
+                                  n_jobs=self.n_jobs, fit_params=fit_params)
+
+        self.base_estimator.fit(X, y, **fit_params)
+
+        if preds.ndim == 1:
+            preds = preds.reshape(-1, 1)
+
+        return preds
+
+
+def _identity(x):
+    return x
+
+
+def _identity_transformer():
+    """Contructs a transformer that returns its input unchanged"""
+    return FunctionTransformer(_identity, accept_sparse=True)
+
+
+def make_stack_layer(base_estimators, restack=False, cv=3, method='auto',
+                     n_jobs=1, n_cv_jobs=1, transformer_weights=None):
+    """ Construct single layer for model stacking
+
+    Parameters
+    ----------
+    base_estimators : list of estimators to be used in stacking
+
+    restack : bool, optional (default=False)
+        Whether input should be concatenated to the transformation.
+
+    cv : cv to be used, optional (default=3)
+        Will be passed to ``StackingTransformer`` for each base estimator.
```

Comment:
Copy the description from StackingTransformer here rather than referring the user.